scm/driver/gitea/git.go

```diff
@@ -30,6 +30,17 @@ func (s *gitService) FindCommit(ctx context.Context, repo, ref string) (*scm.Com
 	return convertCommitInfo(out), res, err
 }
 
+func (s *gitService) FindCommitByRef(ctx context.Context, repo, ref string) (*scm.Commit, *scm.Response, error) {
+	path := fmt.Sprintf("api/v1/repos/%s/git/%s", repo, ref)
+	out := []*refInfo{}
+	_, err := s.client.do(ctx, "GET", path, nil, &out)
+    if err != nil {
+        return nil, nil, err
+    }
+    sha := out[0].Object.Sha
```

Comment:
this code is subject to a panic if the index is out of bounds, so we should probably check the length of the slice just to be certain.